.github/workflows/hq-smoke-tests.yml

```
@@ -12,120 +12,128 @@ on:
-        from: <${{secrets.DIMAGIQA_MAIL_USERNAME}}>
-        html_body: file:////home/runner/work/dimagi-qa/dimagi-qa/HQSmokeTests/email_fail.html
-        attachments: /home/runner/work/dimagi-qa/dimagi-qa/report.html
-        priority: normal
+      - name: Send Failure Email on Push or Pull
+        uses: dawidd6/action-send-mail@v3
+        if: ${{ failure() && github.event_name != 'repository_dispatch' }}
+        with:
+          server_address: smtp.gmail.com
+          server_port: 465
+          username: ${{secrets.DIMAGIQA_MAIL_USERNAME}}
+          password: ${{secrets.DIMAGIQA_MAIL_PASSWORD}}
+          subject: FAIL - Smoke Tests[#${{github.run_number}}] for ${{ matrix.environment }} on branch "${{ github.head_ref }}", ${{env.NOW}}
+          to: qa-automation@dimagi.com
+          from: <${{secrets.DIMAGIQA_MAIL_USERNAME}}>
+          html_body: file:////home/runner/work/dimagi-qa/dimagi-qa/HQSmokeTests/email_fail.html
+          attachments: /home/runner/work/dimagi-qa/dimagi-qa/report_${{ matrix.environment }}.html
+          priority: normal
 
-    - name: Send Pass Email on Push or Pull
-      uses: dawidd6/action-send-mail@v3
-      if: ${{ success() && github.event_name != 'repository_dispatch' }}
-      with:
-        server_address: smtp.gmail.com
-        server_port: 465
-        username: ${{secrets.DIMAGIQA_MAIL_USERNAME}}
-        password: ${{secrets.DIMAGIQA_MAIL_PASSWORD}}
-        subject: PASS - Smoke Tests[#${{github.run_number}}] on branch "${{ github.head_ref }}", ${{env.NOW}}
-        to: qa-automation@dimagi.com
-        from: <${{secrets.DIMAGIQA_MAIL_USERNAME}}>
-        html_body: file:////home/runner/work/dimagi-qa/dimagi-qa/HQSmokeTests/email_pass.html
-        priority: normal
+      - name: Send Pass Email on Push or Pull
+        uses: dawidd6/action-send-mail@v3
+        if: ${{ success() && github.event_name != 'repository_dispatch' }}
+        with:
+          server_address: smtp.gmail.com
+          server_port: 465
+          username: ${{secrets.DIMAGIQA_MAIL_USERNAME}}
+          password: ${{secrets.DIMAGIQA_MAIL_PASSWORD}}
+          subject: PASS - Smoke Tests[#${{github.run_number}}] for ${{ matrix.environment }} on branch "${{ github.head_ref }}", ${{env.NOW}}
+          to: qa-automation@dimagi.com
+          from: <${{secrets.DIMAGIQA_MAIL_USERNAME}}>
+          html_body: file:////home/runner/work/dimagi-qa/dimagi-qa/HQSmokeTests/email_pass.html
+          priority: normal
 
-    - name: Archive test results
-      if: always()
-      uses: actions/upload-artifact@v2
-      with:
-        name: test-result-reports
-        path: /home/runner/work/dimagi-qa/dimagi-qa/report.html
-        retention-days: 2
+      - name: Archive test results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-result-reports-${{ matrix.environment }}
+          path: /home/runner/work/dimagi-qa/dimagi-qa/report_${{ matrix.environment }}.html
```

Rather use the provided workspace path
```suggestion
          path: ${{ github.workspace }}/report_${{ matrix.environment }}.html
```